edb/pgsql/compiler/relctx.py

```diff
@@ -85,21 +96,31 @@ def pull_path_namespace(
             # Skip pulling paths that match the path_id_mask before or after
             # doing path id mapping. We need to look at before as well
             # to prevent paths leaking out under a different name.
-            if (
+            if flavor == 'normal' and (
```

Comment:
I'd invert the check to `flavor != 'packed'` and add a comment why `path_id_mask` doesn't apply for it.